### ISSUE TITLE
monaco-languageclient version upgrade needed to support latest monaco-editor 0.36.0

### ISSUE THREAD

**Issue by OliverwengFiltered** [TypeFox/monaco-languageclient#465]
Hi folks,

I'm wondering if anyone could upgrade the monaco-languageclient so that it can build with the latest monaco-editor

Currently it complains about some of the exports not available in a bunch of monaco editor node_module files, that means the latest monaco-languageclient can't be built with latet monaco-editor version 0.36.0

Help? idea? insights?
Thanks

**Comment by CGNonofr** [comment]
The update to the ~~0.35~~ 0.36 is planned, don't worry, it just takes a little more time to find a resource to do it.

**Comment by kaisalmen** [comment]
Tha helps to understand what happend 😆 
![image](https://user-images.githubusercontent.com/21079138/221510975-723dba3e-7749-4226-8c0c-d848d9372f01.png)

**Comment by samirzubi-db** [comment]
@CGNonofr any ETA when can we expect the update to be published? Thanks!

**Comment by CGNonofr** [comment]
I don't think I'll be able to provide any ETA, I'll try ASAP. Feel free to help if you have some time

**Comment by kaisalmen** [comment]
@CGNonofr release two pre-release of the next version of `monaco-vscode-api`
I have created https://github.com/TypeFox/monaco-languageclient/tree/update-monaco-vscode-1.76 here and want to put up a new 5.0.0 pre-release soon.

**Comment by kaisalmen** [comment]
Here you go:
https://www.npmjs.com/package/monaco-languageclient/v/5.0.0-next.4

**Comment by kaisalmen** [comment]
The latest changes of `monaco-vscode-api` need to be ready and released before the next release. Sorry for the delay, but it is worth it.

**Comment by kaisalmen** [comment]
#470 is merged. Documentation needs to be aligned, expanded, corrected now.

**Comment by OliverwengFiltered** [comment]
I gave the latest version  a try this morning, however, I ran into this issue:

```
Can't resolve 'vscode' in '/Users/xxx/xxx/xxx/node_modules/vscode-languageclient/lib/common'
 @ ./node_modules/monaco-languageclient/lib/monaco-language-client.js 35:0-89 126:33-53
 @ ./node_modules/monaco-languageclient/lib/index.js 5:0-44 5:0-44
```

monaco-editor version: 0.36.1
monaco-languageclient: 5.0.1
vscode-ws-jsonrpc: 3.0.0

This old problem seems showing up again, any suggestions to solve ?

I have tried to add alias under resolve section for webpack.config.js, it only leads to another issue says `Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './lib/vscode-compatibility' is not defined by "exports"`

**Comment by kaisalmen** [comment]
@OliverwengFiltered no, that should no be required. In a new fresh npm project with only monaco-languagclient as dependency. `npm list vscode` returns this:
![image](https://user-images.githubusercontent.com/21079138/231179830-0cbe34c3-3334-4d50-9b56-2c29586b5f5a.png)
Do you use **npm** or something else. Did you remove your `package-lock.json` for testing and re-install everything?

**Comment by OliverwengFiltered** [comment]
@kaisalmen 
I did try remove my yarn.lock file, it's the same result.
I'm using `yarn` for my project btw.

**Comment by kaisalmen** [comment]
We defined the following peerDependency and npm automatically resolves this dependency in a project using `monaco-languagclient` as dependency. I assume yarn behaves the same. What version of yarn do you use? Did you investigate if yarn treats peerDependecies the same way as npm?
```json
"peerDependencies": {
  "vscode": "npm:@codingame/monaco-vscode-api@~1.76.6"
}
```

**Comment by OliverwengFiltered** [comment]
@kaisalmen 

It's odd, I did some investigation, I can't find a reason why when using yarn to install latest monaco-languageclient it removes the vscode peerDependencies.

but I think it should be reproducible on your end. it might be some code pushed up during the release of the latest monaco-languageclient that's preventing yarn to add the peerDependencies ?

I also tried to add it manually, but code throws error on the run time. it complains about the `this._encounteredLanguages` is undefined.

![Screenshot 2023-04-12 at 10 30 14 AM](https://user-images.githubusercontent.com/28814939/231490210-2c89c422-5331-4f05-bcad-37c8c6142256.png)
![Screenshot 2023-04-12 at 10 30 06 AM](https://user-images.githubusercontent.com/28814939/231490212-eb2d8564-8f3b-4939-bf48-da070167f813.png)

**Comment by abdalla-rko** [comment]
> @kaisalmen yes, I tried add that into my package.json, it built successfully, but the code throws the above error during the run time.

What version of Monaco Editor do you have? I had the same issue when I upgraded to 0.37.1, but when I downgraded to v0.36.1, it worked fine. I just had to add the yarn workaround in the package.json.

**Comment by OliverwengFiltered** [comment]
@abdalla-rko 

Yes, confirmed, it's all working with monaco-editor 0.36.1, thank you, all.
